lectures/12-strings/code/src/com/kunal/Comparison.java

```diff
@@ -0,0 +1,20 @@
+package com.kunal;
+
+public class Comparison {
+    public static void main(String[] args) {
+        String a = "Kunal";
+        String b = "Kunal";
+        String c = a;
+//        System.out.println(c == a);
+        // ==
+//        System.out.println(a == b);
+
+        String name1 = new String("Kunal");
+        String name2 = new String("Kunal");
+
+        System.out.println(name1 == name2);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.